server/src/main/java/org/opensearch/action/search/SearchResponse.java

```diff
@@ -604,6 +673,68 @@ public String toString() {
         }
     }
 
+    /**
+     * Holds info about the clusters that the search was executed on: how many in total, how many of them were successful
+     * and how many of them were skipped.
+     *
+     * @opensearch.internal
+     */
+    public static class PhaseTook implements ToXContentFragment, Writeable {
+        static final ParseField PHASE_TOOK = new ParseField("phase_took");
+        private final Map<String, Long> phaseTookMap;
+
+        public PhaseTook(Map<String, Long> phaseTookMap) {
+            this.phaseTookMap = phaseTookMap;
+        }
+
+        private PhaseTook(StreamInput in) throws IOException {
+            this(in.readMap(StreamInput::readString, StreamInput::readLong));
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeMap(phaseTookMap, StreamOutput::writeString, StreamOutput::writeLong);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject(PHASE_TOOK.getPreferredName());
+
+            for (SearchPhaseName searchPhaseName : SearchPhaseName.values()) {
+                if (phaseTookMap.containsKey(searchPhaseName.getName())) {
+                    builder.field(searchPhaseName.getName(), phaseTookMap.get(searchPhaseName.getName()));
+                } else {
+                    builder.field(searchPhaseName.getName(), 0);
+                }
+            }
+            builder.endObject();
+            return builder;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
```

Comment:
Sorry for missing this earlier -- I was just scrutinizing the output from Codecov to see if any of code not covered by tests have any risks, and I noticed this `equals` implementation.

Can you please add a unit test that creates two PhaseTook instances, like:

```
PhaseTook phaseTookA = new PhaseTook(Map.of("foo", 0, "bar", 1));
PhaseTook phaseTookB = new PhaseTook(Map.of("foo", 0));

assertNotEqual(phaseTookA, phaseTookB);
assertNotEqual(phaseTookB, phaseTookA);
```

I think the first assertion will succeed, but the second one will fail.

Reply:
Good catch, fixed it and added a test